site/source/docs/porting/exceptions.rst

```diff
@@ -146,6 +148,11 @@ To use this function, you need to pass ``-sEXPORT_EXCEPTION_HANDLING_HELPERS``
 to the options. You need to enable either of Emscripten EH or Wasm EH to use
 this option.
 
+If the stack pointer has been moved due to stack allocations within the Wasm
+function before an exception is thrown, you can to use ``stackSave()`` and
+``stackRestore()`` to restore the stack pointer so that no stack memory is
+leaked.
```

Comment:
What about C++ destructors on the stack?  Are they run in this scenario?  A assume yes.

Reply:
Does it? If the whole `try`-`catch` is in C++ it would have run the destructor for the stack object at the point we go out of the `try` scope, but in this case it is JS, no?

Reply:
I assume that each stack from that contains destructors has it own code the runs during unwind, before the exception escapes the JS, is that not how it works?

Reply:
Ah right I was confused. The destructor would have run, but the stack pointer is not adjusted back, because we do it within the `catch` block of the wasm code.